src/main/java/org/carlmontrobotics/lib199/MotorControllerFactory.java

```diff
@@ -78,86 +87,184 @@ public static WPI_TalonSRX createTalon(int id) {
     return talon;
   }
 
-  //checks for spark max errors
-
-  @Deprecated
-  public static CANSparkMax createSparkMax(int id, MotorErrors.TemperatureLimit temperatureLimit) {
-    return createSparkMax(id, temperatureLimit.limit);
-  }
-
-  @Deprecated
-  public static CANSparkMax createSparkMax(int id, int temperatureLimit) {
-    CANSparkMax spark;
+  /**
+   * Create a default sparkMax controller (NEO or 550).
+   * 
+   * @param id the port of the motor controller
+   * @param motorConfig either MotorConfig.NEO or MotorConfig.NEO_550
+   */
+  public static SparkMax createSparkMax(int id, MotorConfig motorConfig) {
+    SparkMax spark=null;
     if (RobotBase.isReal()) {
-      spark = new CANSparkMax(id, CANSparkLowLevel.MotorType.kBrushless);
+      spark = new SparkMax(id, SparkLowLevel.MotorType.kBrushless);
     } else {
-        spark = MockSparkMax.createMockSparkMax(id, CANSparkLowLevel.MotorType.kBrushless);
+      spark = MockSparkMax.createMockSparkMax(id, SparkLowLevel.MotorType.kBrushless, MockSparkMax.NEOType.NEO);
     }
-    spark.setPeriodicFramePeriod(CANSparkLowLevel.PeriodicFrame.kStatus0, 1);
 
-    MotorErrors.reportSparkMaxTemp(spark, temperatureLimit);
+    // config.setPeriodicFramePeriod(SparkLowLevel.PeriodicFrame.kStatus0, 1);
+    if (spark!=null)
+      MotorErrors.reportSparkMaxTemp(spark, motorConfig.temperatureLimitCelsius);
+    
+    MotorErrors.checkSparkMaxErrors(spark);
 
-    MotorErrors.reportError(spark.restoreFactoryDefaults());
-    MotorErrors.reportError(spark.follow(ExternalFollower.kFollowerDisabled, 0));
-    MotorErrors.reportError(spark.setIdleMode(IdleMode.kBrake));
-    MotorErrors.reportError(spark.enableVoltageCompensation(12));
-    MotorErrors.reportError(spark.setSmartCurrentLimit(50));
+    if (motorConfig==MotorConfig.NEO || motorConfig==MotorConfig.NEO_550)
+      spark.configure(baseSparkMaxConfig(), SparkBase.ResetMode.kResetSafeParameters, SparkBase.PersistMode.kPersistParameters);
+    else if (motorConfig==MotorConfig.NEO_VORTEX)
+      spark.configure(baseSparkFlexConfig(), SparkBase.ResetMode.kResetSafeParameters, SparkBase.PersistMode.kPersistParameters);
+    else
+      spark.configure(baseSparkConfig(), SparkBase.ResetMode.kResetSafeParameters, SparkBase.PersistMode.kPersistParameters);
 
-    MotorErrors.checkSparkMaxErrors(spark);
 
-    SparkPIDController controller = spark.getPIDController();
-    MotorErrors.reportError(controller.setOutputRange(-1, 1));
-    MotorErrors.reportError(controller.setP(0));
-    MotorErrors.reportError(controller.setI(0));
-    MotorErrors.reportError(controller.setD(0));
-    MotorErrors.reportError(controller.setFF(0));
+    return spark;
+  }
+  /**
+   * Create a sparkMax controller (NEO or 550) with custom settings.
+   * 
+   * @param id the port of the motor controller
+   * @param config the custom config to set
+   */
+  public static SparkMax createSparkMax(int id, SparkBaseConfig config) {
+    SparkMax spark = null;
```

Comment:
I'd get rid of the `null` to make it clear we're always defining it.
```suggestion
    SparkMax spark;
```